### PR TITLE
[C#] feat: Streaming + Sample

### PR DESCRIPTION
**Pull request by lilyydu**
## Linked issues

closes: #1968 

## Details

Added streaming support for C# with an associated sample bot.

#### Change details

- created a separate `IPromptCompletionStreamingModel` (not possible to have optional members for an interface). As a result, OpenAIModel declaration had to be updated.
- converted `StreamingChannelData` into a class

## Attestation Checklist
- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/IPromptCompletionModelEvents.cs [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    /// Events emitted by a IPromptCompletionStreamingModel.
+    /// </summary>
+    public interface IPromptCompletionModelEvents
+    {
+
+        /// <summary>
+        /// Defines the method that is triggered before the model is called to complete a prompt.
+        /// </summary>
+        /// <returns></returns>
+        public delegate void BeforeCompletionHandler(object sender, BeforeCompletionEventArgs args);
```
Comment (singhk97) [inline dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/IPromptCompletionModelEvents.cs:19]: Is there a reason `BeforeCompletionEventArgs` expects a cancellation token? It should be passed only if the delegate is an "async" delegate ... i.e it returns a `Task` object.

Also the convention is to pass the cancellation token as the last parameter of the delegate. Here's an example: https://github.com/microsoft/teams-ai/blob/e4a556b65892920c171fdbccdb6a8ce7a895c596/dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/Application/AdaptiveCards/AdaptiveCardsHandlers.cs#L74

**Review by singhk97 — COMMENTED** [review]
LGTM!

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/IPromptCompletionModelEvents.cs (hunk 1 above)
Comment (lilyydu) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/IPromptCompletionModelEvents.cs:19]: Discussed and removed, forgot to delete

**Review by aacebo — COMMENTED** [review]
is there a way for us to update the existing chefbot and make streaming configurable? My thought is that instead of having samples for streaming and non streaming, we should treat it like we do with OpenAI/AzureOpenAI, where every samples supports it but its a configuration

**Review by aacebo — APPROVED** [review]

**Pull request merged**